kubestash/__init__.py

```diff
@@ -396,14 +491,87 @@ def cmd_push(args):
                                                                                              secret=args.secret))
 
 
-def cmd_daemon(args):
-    # https://boto3.readthedocs.io/en/latest/reference/services/dynamodb.html
-    # https://boto3.readthedocs.io/en/latest/reference/services/dynamodbstreams.html
-    # TODO: what if there are more than 100 streams?
+def cmd_pushall(args):
+    """Syncs a Credstash table with an entire cluster"""
 
-    # -f --force is implied by daemon
-    args.force = True
+    # Map of all namespaces to secrets
+    secretMap = {}
+
+    # Depending on whether or not we need to
+    if args.secretname:
+        # If the string contains slashes
+        # we can infer the namespace and secretname
+        _secretname = args.secretname
+        secretkey = args.secretname
+        if "/" in args.secretname:
+            args.namespace, args.secret, _secretname = args.secretname.split('/')
+        else:
+            secretkey = args.namespace + "/" + args.secret + "/" + args.secretname
+
+        if args.verbose:
+            print("Syncing a single secret key: key={key}, ns={namespace}, secret={secret}".format(key=secretkey, namespace=args.namespace, secret=args.secret))
+
+        secrets = {args.secretname: credstash_getone(secretkey, args)}
 
+        secretMap = {
+            args.namespace: set([args.secret])
+        }
+    else:
+        prefix = ''
+
+        # This is incorrect, since it doesn't work for the `default` namespace
+        # TODO: Figure out a way!
+        if args.namespace != 'default':
```

Comment:
The default value here causes an issue, since `pushall` is supposed to run on all namespaces. The issue arises because `--namespace` is defined in the base parser, and cannot be overwritten (I didn't try much) in just the pushall command. It should be optional, but have a default value of `None` so as to avoid this edgecase.